src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/pagepart/Resources/views/PageParts/CroppableImagePagePart/view.html.twig

```diff
@@ -0,0 +1,46 @@
+{% import _self as buildImage %}
+
+{% macro setFilters(media, altText, type, originalWidth) %}
+
+    {% set imgUrl = '' %}
+
+    {% set imgUrl_huge = cropped_imagine_filter(media, 'desktop') %}
+    {% set imgUrl_big = cropped_imagine_filter(media, 'desktop') %}
+    {% set imgUrl_medium = cropped_imagine_filter(media, 'tablet') %}
+    {% set imgUrl_small = cropped_imagine_filter(media, 'phone') %}
+    {% set srcSet =  imgUrl_small ~ ' 400w, ' ~ imgUrl_medium ~ ' 600w, ' ~ imgUrl_big ~ ' 1200w, ' ~ imgUrl_huge ~ ' 2400w' %}
+
+    <img srcset="{{ srcSet }}" sizes="100vw"{% if altText %} alt="{{ altText }}"{% endif %} width="{{ originalWidth }}"/>
+    <noscript>
+        <img src="{{ imgUrl_big }}"{% if altText %} alt="{{ altText }}"{% endif %} />
+    </noscript>
+{% endmacro %}
+
+{% if resource.media is not empty %}
+    {% set image %}
+        {% set imageType = resource.media.media.originalFilename|lower|split('.')|last %}
+        {{ buildImage.setFilters(resource.media, resource.altText, imageType, resource.media.media.metadata.original_width) }}
+    {% endset %}
+
+    {% if resource.media is not empty %}
+        {% if app.request %}
+            <figure class="image-pp">
+                {% set imageType = resource.media.media.originalFilename|lower|split('.')|last %}
+
+                {% if resource.link is defined and resource.link != '' %}
+                    <a href="{{ resource.link | replace_url }}" {% if resource.openinnewwindow %}target="_blank" rel="noopener"{% endif %}>
+                        {{ image }}
+                    </a>
+                {% else %}
+                    {{ image }}
+                {% endif %}
+
+                {% if resource.caption %}
```

Comment:
`resource.caption` might not always be declared in the pagepart class, add a `is defined` check